### PR TITLE
#7304 Aligned standard template to latest changes

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
Need to do a final test. 

Anyway I proceeded this way:
- I generated a project, applied fixes and ported them in template files
- Did a diff of the previous template files and the current ones to create the migration guidelines, that are long but well detailed

Tests to do:
 - Create a project and verify that project work, and that the application can work with extensions
 - Setup data dir and check if everything works as expected, as for the product.

I noticed that some of the fixes are missing also on master, so we should do a similar PR on master (also to port the Migration guidelines). But the things changes because of all updated to geostore, so we need a different PR. 
If on the next release we use the new project system, doing this additional work may become obsolete.
 
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#7304

**What is the new behavior?**
Project created with `createProject now are working, and we can install extensions in them
Migration guidelines has been updated, so we can update existing project with the new features.

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: project/standard/templates/web/src/main/webapp/WEB-INF/web copy.xml [hunk 1]
```diff
@@ -0,0 +1,143 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mbarto) [inline project/standard/templates/web/src/main/webapp/WEB-INF/web copy.xml:1]: commited by mistake?

Comment (offtherailz) [reply project/standard/templates/web/src/main/webapp/WEB-INF/web copy.xml]: yes thank you

File: docs/developer-guide/mapstore-migration-guide.md [hunk 2]
```diff
@@ -68,29 +67,293 @@ In the future the current script will be deprecated in favor of the new one.
...
+```
 
 #### Back-end has been reorganized
 
 In particular:
 
 - all the java code has been moved from `web/src/` to the `java/` and `product/` directories (and `release`, already existing).
 - `mapstore-backend` has been renamed into `mapstore-services`.
-- `geostore` version used is `1.7.0` on release `2021.02.00`
+- Some servlets has been added in order to provide native support to data dir and make it work with the new `configs` directory.
```
Comment (mbarto) [inline docs/developer-guide/mapstore-migration-guide.md:189]:
```suggestion
- Some servlets have been added in order to provide native support to data dir and make it work with the new `configs` directory.
```

**Review by mbarto — APPROVED** [review]
LGTM, just one trivial suggestion

**Comment by tdipisa** [comment]
@offtherailz this need to be tested. Please organize with @ElenaGallo is possible. We have another dedicated issue to do it for master.

**Comment by ElenaGallo** [comment]
Test passed, @offtherailz please backport to 2021.02.xx

**Comment by offtherailz** [comment]
@ElenaGallo this was an exception. It had to be tested on 2021.02.xx, then ported on master, because they require 2 different PRs. 
Did you tested on master ?

**Comment by ElenaGallo** [comment]
sorry @offtherailz, you're right I tried it on the 2021.02.xx branch. I can close the issue. Thank you

**Pull request merged**